main.go

```diff
@@ -128,10 +341,161 @@ func determineShouldFetch(lastFetchedFilepath string) bool {
 		lastFetchedTimestampUintParseBits,
 	)
 	if err != nil {
-		logrus.Errorf("An error occurred parsing last-fetch Unix time string '%v':\n%v", err)
+		return false, stacktrace.Propagate(err, "An error occurred parsing last-fetch Unix time string: '%v'.", lastFetchedUnixTimeStr)
 	}
 	lastFetchedTime := time.Unix(int64(lastFetchedUnixTime), extraNanosecondsToAddToLastFetchedTimestamp)
 	noFetchNeededBefore := lastFetchedTime.Add(fetchGracePeriod)
 
-	return time.Now().After(noFetchNeededBefore)
+	return time.Now().After(noFetchNeededBefore), nil
 }
+
+func getLatestReleaseVersion(repo *git.Repository) (*semver.Version, error) {
+	tagrefs, err := repo.Tags()
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred while retrieving tags for repository.")
+	}
+
+	// Trim tagrefs and filter for only tags with X.Y.Z version format
+	var allTagSemVers []*semver.Version
+	err = tagrefs.ForEach(func(tagref *plumbing.Reference) error {
+		tagName := tagref.Name().String()
+		tagName = strings.ReplaceAll(tagName, tagsPrefix, "")
+
+		if semverRegex.Match([]byte(tagName)) {
+			tagSemVer, err := semver.StrictNewVersion(tagName)
+			if err != nil {
+				return stacktrace.Propagate(err, "An error occurred while retrieving the following tag: %s.", tagName)
```

Comment:
Error message doesn't match what it's doing; this is actually parsing a tag name to a semver object (also same `:`/`''`/`.` thing)